lib/web_ui/lib/src/engine/context_menu.dart

```diff
@@ -0,0 +1,37 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dom.dart';
+import 'safe_browser_api.dart';
+
+/// Manages the browser's context menu for the entire app.
+///
+/// The context menu is the menu that appears on right click, for example.
+class ContextMenu {
+  const ContextMenu._();
+
+  /// Handler for contextmenu events that prevents the browser's context menu
+  /// from being shown.
+  static final DomEventListener _handleContextMenu = allowInterop((DomEvent event) {
+    event.preventDefault();
+  });
+
+  /// Disables the browser's context menu.
+  ///
+  /// By default, when a Flutter web app starts, the context menu is enabled.
+  ///
+  /// Can be re-enabled by calling [enableContextMenu].
+  static void disableContextMenu() {
+    domWindow.addEventListener('contextmenu', _handleContextMenu);
```

Comment:
The flutter app could be embedded inside a host element (instead of taking over the entire window). We should add the event listener to the host element. cc @ditman for the right way to do this.

Reply:
The `contextmenu` event bubbles ([spec](https://w3c.github.io/uievents/#event-type-contextmenu)), so this should be enough?

If this needs to be aware of the embedding, the enable/disable methods could be added to the `EmbeddingStrategy` abstract class (and its implementations): 

https://github.com/flutter/engine/blob/main/lib/web_ui/lib/src/engine/view_embedder/embedding_strategy/embedding_strategy.dart#L23

Reply:
> The `contextmenu` event bubbles ([spec](https://w3c.github.io/uievents/#event-type-contextmenu)), so this should be enough?

Yes, but it's more than enough :) It disables the context menu for the entire page, which we don't want in the host element mode.

Reply:
You're very right :P

Reply:
Thanks for the advice here!  I've moved the enable/disable methods to the embedding strategy and made them accessible via FlutterViewEmbedder.